my-new-webapp-service/src/main/java/service/impl/CustomerServiceImpl.java

```diff
@@ -51,7 +51,6 @@ public Customer updateCustomer(Customer customer, Integer id) throws EntityNotFo
             throw new EntityNotFoundException("Was not been updated any rows");
         }
 
-        customer.setId(id);
-        return customer;
+        return cd.getCustomer(customer.getLogin(), customer.getPassword());
```

Comment:
Excuse me, but I have not understand what you mean exactly. When I wrote **updateCustomer()** method I thought that it will be used when user want to update himself data in profile, where he should write login and old password. Maybe you suggest to change signature of the method to use it when user forgot password or lost it? Thanks :)

Reply:
I mean that I want you to make a Data Base do exact amount of job that she need to, no less no more. Let's talk about it from the performance side. Making searches must be done only on the indexed fields. Just because on a vast amount of data that might be tens times faster. For us these fields are id (because of being PK) and login (because of being UNIQUE).

Reply:
Besides, I've never seen a web-site where I could change login and password (and even a name) in the exact same form. Just imagine how weird is that. That's might be not changed, suits for study proj, but for the future times take it to account.